app/src/main/java/com/codesoom/assignment/interceptors/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,68 @@
+package com.codesoom.assignment.interceptors;
+
+import com.codesoom.assignment.application.authentication.AuthenticationService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@RequiredArgsConstructor
+@Component
+public class AuthenticationInterceptor implements HandlerInterceptor {
+    private final AuthenticationService authenticationService;
+
+    @Override
+    public boolean preHandle(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            Object handler
+            ) throws Exception {
+        return filterWithPathAndMethod(request) ||
+                doAuthentication(request, response);
+    }
+
+    private boolean filterWithPathAndMethod(HttpServletRequest request) {
+        String path = request.getRequestURI();
+        String method = request.getMethod();
+        if (path.startsWith("/products") && (
+                method.equals("GET") || method.equals("OPTIONS"))
+        ) {
+            return true;
+        }
+
+        if ((path.startsWith("/session") || path.startsWith("/users"))
+                && method.equals("POST")) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * 유효한 토큰으로 요청을 했는지 확인
```

Comment:
boolean 메소드는 이런 방식으로 JavaDoc을 작성해 보세요.

```suggestion
     * 유효한 토큰으로 요청했다면 true, 그렇지 않다면 false를 리턴합니다.
```

Reply:
boolean을 반환하는 메소드는 어떤상황에서 true와 false가 발생하는지를 작성하는 것이 좋은 작성방법 이네요!
만약 Optional 을 반환하는 경우에는 `Optional.empty` 와 `Optional.of` 인 두가지의 경우를 서술하는 것이 좋을까요?

Reply:
Optional이라면 그렇게까지 하지 않아도 괜찮아요. 하지만 설명이 필요한 경우도 있으니 양쪽을 잘 저울질해봐야겠죠.